src/c/funkce.md

```diff
@@ -201,11 +201,45 @@ funkce). Takovýhle kód pak lze číst téměř jako větu v přirozeném jazyc
 - Sdílení kódu: pokud budete chtít použít kód, který napsal někdo jiný, tak toho dosáhnete právě
 používáním funkcí, které vám někdo [připraví](knihovny.md).
 
-### Umístění funkcí
+## Umístění funkcí
 Funkce v *C* musíme psát vždy na nejvyšší úrovni souboru. V *C* tedy například není možné definovat
 funkci uvnitř jiné funkce:
 ```c,editable,readonly
 int main() {
     int test() { }
 }
 ```
+
+## Proč název "funkce"?
+Možná vás napadlo, že název funkce zní podobně jako [funkce](https://matematika.cz/co-je-to-funkce)
+v matematice. Není to náhoda, funkce v programech se tak opravdu dají částečně chápat – berou nějaký
+vstup (parametry) a vracejí výstup (návratovou hodnotu). Například následující matematickou funkci:
+
+\\( f(x) = 2 * x \\)
+
+můžeme v *C* naprogramovat takto:
+```c
+int f(int x) {
+    return 2 * x;
+}
+```
+Aby ale funkce v *C* splňovala požadavky matematické funkce, musí být splněno několik vlastností:
+- Funkce nesmí mít žádné [vedlejší efekty](prikazy_vyrazy.md#vedlejší-efekty). To znamená, že by
+měla pouze provést výpočet na základě vstupních parametrů a vrátit vypočtenou hodnotu. Neměla by
+číst nebo modifikovat [globální proměnné](globalni_promenne.md) nebo například interagovat se
+soubory na disku.
+- Funkce musí mít návratový typ jiný než `void`, aby vracela nějakou hodnotu. Z toho také vyplývá,
+že funkce s návratovým typem `void` musí mít nutně nějaké vedlejší efekty, jinak by totiž nemělo
+cenu ji volat (protože nic nevrací).
+- Pokud je funkce zavolána se stejnými hodnotami parametrů, musí vždy vrátit stejnou návratovou
+hodnotu. Této vlastnosti se říká *idempotence*. Jelikož jsou počítače deterministické, tato
+vlastnost by měla být triviálně splněna, pokud funkce neobsahuje žádné vedlejší efekty.
+
+Funkce splňující tyto vlastnosti se nazývají *čisté* (*pure*). S takovýmito funkcemi je jednodušší
+pracovat a přemýšlet nad tím, co dělají, protože si můžeme být jistí, že nemodifikují okolní stav
+programu a pouze spočítají výsledek v závislosti na svých parametrech. Pokud to tedy jde, snažte se
+funkce psát tímto stylem (samozřejmě ne vždy je to možné).
+
+V předmětu
+[Funkcionální programování](https://edison.sso.vsb.cz/cz.vsb.edison.edu.study.prepare.web/SubjectVersion.faces?version=460-2054/01)
```

Comment:
Nepridame i tenhle link: http://behalek.cs.vsb.cz/wiki/index.php/Functional_programming/cs ? Sice se to muze v case menit, ale i tak by to mozna slo.

Reply:
Jj, dám ho tam.